umbra-js/src/classes/Umbra.ts

```diff
@@ -313,107 +313,161 @@ export class Umbra {
+    // Graph query failed, try requesting logs directly IF we are not on polygon. If we are on
+    // Polygon, there isn't much we can do, so for now just show a warning and return an empty array
+    if (this.chainConfig.chainId === 137) {
+      throw new Error('Cannot fetch Announcements from logs on Polygon, please try again later');
+    }
+
     // Get list of all Announcement events
     const announcementFilter = this.umbraContract.filters.Announcement(null, null, null, null, null);
-    const announcements = await this.umbraContract.queryFilter(announcementFilter, startBlock, endBlock);
+    const announcementEvents = await this.umbraContract.queryFilter(announcementFilter, startBlock, endBlock);
 
-    const userAnnouncements = await Promise.all(
-      announcements.map(async (event) => {
+    const announcements = await Promise.all(
+      announcementEvents.map(async (event) => {
         // Extract out event parameters
         const announcement = (event.args as unknown) as Announcement;
-        const { receiver, amount, token } = announcement;
-        const { isForUser, randomNumber } = isAnnouncementForUser(spendingPublicKey, viewingPrivateKey, announcement);
+        const { receiver, amount, token, ciphertext, pkx } = announcement;
 
-        // If  receiving address matches event's recipient, the transfer was for the user. Otherwise it wasn't,
-        // so return null and filter later
-        if (!isForUser) return null;
         const [block, tx] = await Promise.all([event.getBlock(), event.getTransaction()]);
         return {
-          randomNumber,
-          receiver,
           amount,
-          token: getAddress(token),
+          block: block.number.toString(),
+          ciphertext,
           from: tx.from,
-          txHash: event.transactionHash,
+          receiver,
+          pkx,
           timestamp: String(block.timestamp),
-          isWithdrawn: false,
+          token: getAddress(token),
+          txHash: event.transactionHash,
         };
       })
     );
 
-    return userAnnouncements.filter((ann) => ann !== null) as UserAnnouncement[];
+    return announcements;
+  }
+
+  /**
+   * @notice Scans all Umbra event logs for funds sent to the specified address
+   * @dev This is a convenience method that first runs `fetchAllAnnouncements`, then filters them through
+   * the static helper `isAnnouncementForUser`. The latter is CPU intensive and this method will block while
+   * all announcements are processed. To avoid performance issues, you may need to run fetching and filtering
+   * steps separately, and use chunking, web workers, or other threading strategies.
+   * @param spendingPublicKey Receiver's spending private key
+   * @param viewingPrivateKey Receiver's viewing public key
+   * @param overrides Override the start and end block used for scanning
+   */
+  async scan(spendingPublicKey: string, viewingPrivateKey: string, overrides: ScanOverrides = {}) {
```

Comment:
just to confirm—this is the same behavior as the original `scan` method, just refactored to use the new methods?

Reply:
That's right! Behaves the same as the original method— I didn't even change the tests. It just leverages the new methods rather than implementing the behavior internally.